### PR TITLE
feat(react-ui-ag): give ListingCarousel an `onIndexChange` prop

### PR DESCRIPTION
**Pull request by nray621**
affects: @rentpath/react-ui-ag
[card](https://rentpath.leankit.com/card/595390934)

* fire onIndexChange in `listingClickHandler` if listing is not currently selected
* add test for `listingClickHandler`
* prevent prioritizeCardClick from being passed through spread props
* Generate new test snap for ListingCarousel-test

### CODE REVIEW

**Comment by paudyalhari** [comment]
👍

**Comment by danthamotheram** [comment]
Code LGTM 👍

File: packages/react-ui-ag/src/Carousels/ListingCarousel.js [hunk 1]
```diff
@@ -43,8 +44,11 @@ export default class ListingCarousel extends Component {
 
   @autobind
   listingClickHandler(index) {
+    const { onIndexChange, listings } = this.props
+
     if (index !== this.state.selectedIndex) {
       this.updatedSelectedIndex(index)
+      if (onIndexChange) onIndexChange(listings[index])
```
Comment (mneumark) [inline packages/react-ui-ag/src/Carousels/ListingCarousel.js]: In the carousel component in core this is called ‘onSlide’. Can we rename this for consistency?

Comment (nray621) [reply packages/react-ui-ag/src/Carousels/ListingCarousel.js]: Yeah no problem. I like that better actually.

**Pull request merged**